lib/msal-browser/src/interaction_client/NativeInteractionClient.ts

```diff
@@ -231,25 +231,138 @@ export class NativeInteractionClient extends BaseInteractionClient {
-        this.browserStorage.setAccount(accountEntity);
+
+        return homeAccountIdentifier;
+    }
+
+    /**
+     * creates accountEntity
+     * @param request 
+     * @param response 
+     * @param homeAccountIdentifier 
+     * @param idTokenObj 
+     * @returns 
+     */
+    protected createAccountEntity(response: NativeResponse, homeAccountIdentifier: string, idTokenObj: AuthToken, authority: string): AccountEntity {
+
+        return AccountEntity.createAccount(response.client_info, homeAccountIdentifier, idTokenObj, undefined, undefined, undefined, authority, response.account.id);
+    }
+
+    /**
+     * 
+     * @param request 
+     * @param response 
+     * @returns 
+     */
+    generateScopes(request: NativeTokenRequest, response: NativeResponse): ScopeSet {
+        return response.scope ? ScopeSet.fromString(response.scope) : ScopeSet.fromString(request.scope);
+    }
+
+    /**
+     * If PoP token is requesred, records the PoP token if returned from the WAM, else generates one in the browser
+     * @param request 
+     * @param response 
+     */
+    async generatePopAccessToken(response: NativeResponse, request: NativeTokenRequest): Promise<string> {
+        
+        if(request.tokenType === AuthenticationScheme.POP) {
+            /** 
+             * This code prioritizes SHR returned from the native layer. In case of error/SHR not calculated from WAM and the AT 
+             * is still received, SHR is calculated locally
+             */
+        
+            // Check if native layer returned an SHR token
+            if (response.shr) {
+                this.logger.trace("handleNativeServerResponse: SHR is enabled in native layer");
+                return response.shr;
+            }
+
+            // Generate SHR in msal js if WAM does not compute it when POP is enabled
+            const popTokenGenerator: PopTokenGenerator = new PopTokenGenerator(this.browserCrypto);
+            const shrParameters: SignedHttpRequestParameters = {
+                resourceRequestMethod: request.resourceRequestMethod,
+                resourceRequestUri: request.resourceRequestUri,
+                shrClaims: request.shrClaims,
+                shrNonce: request.shrNonce
+            };
+
+            /**
+             * KeyID must be present in the native request from when the PoP key was generated in order for
+             * PopTokenGenerator to query the full key for signing
+             */
+            if (!request.keyId) {
+                throw ClientAuthError.createKeyIdMissingError();
+            }
+            return await popTokenGenerator.signPopToken(response.access_token, request.keyId, shrParameters);
+        } else {
+            return response.access_token;
+        }
+    }
+
+    /**
+     * 
+     * @param request 
+     * @param response 
+     * @param accountEntity 
+     * @param idTokenObj 
+     * @param authority 
+     * @param uid 
+     * @param tid 
+     * @param responseScopes 
```

Comment:
Can you check your input paremeters for the method against the comments here?  Some of these don't line up or are missing in different orders in this change.

Reply:
Addressed the comments, they are outfdated. Thanks for catching this.